plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectDelete.java

```diff
@@ -209,21 +211,21 @@ private static void changeDependentObjectsList(@NotNull DBRProgressMonitor monit
 
     @Override
     public void updateElement(UIElement element, Map parameters) {
-//        if (!updateUI) {
-//            return;
-//        }
-//        final ISelectionProvider selectionProvider = UIUtils.getSelectionProvider(element.getServiceLocator());
-//        if (selectionProvider != null) {
-//            ISelection selection = selectionProvider.getSelection();
-//
-//            if (selection instanceof IStructuredSelection && ((IStructuredSelection) selection).size() > 1) {
-//                element.setText(UINavigatorMessages.actions_navigator_delete_objects);
-//            } else {
-//                DBNNode node = NavigatorUtils.getSelectedNode(selection);
-//                if (node != null) {
-//                    element.setText(UINavigatorMessages.actions_navigator_delete_ + " " + node.getNodeTypeLabel()  + " '" + node.getNodeName() + "'");
-//                }
-//            }
-//        }
+        if (!updateUI) {
+            return;
+        }
+        final ISelectionProvider selectionProvider = UIUtils.getSelectionProvider(element.getServiceLocator());
+        if (selectionProvider != null) {
+            ISelection selection = selectionProvider.getSelection();
+            if (selection instanceof IStructuredSelection) {
+                List<DBNNode> nodes = NavigatorUtils.getSelectedNodes(selection);
+                for (DBNNode node : nodes) {
+                    if (!(node instanceof DBNDataSource)) {
+                        return;
+                    }
+                }
+                element.setText(UINavigatorMessages.actions_navigator_delete_ + " " + nodes.get(0).getNodeTypeLabel());
```

Comment:
Are we sure that we want to update label only for connections?

Reply:
I guess so. The reason to write it is only for connection, because it could be a bit confusing what exactly do you want to delete? A database or connection? But it's clear enough if you try to delete for example a table. You don't need extra explonation for it as well as we don't need to overload our UI